src/upgrade.py

```diff
@@ -86,6 +93,14 @@ def _on_upgrade_charm_check_legacy(self) -> None:
             logger.debug("Wait all units join the upgrade relation")
             return
 
+        if self.charm.unit.is_leader() and self.charm._peers:
+            for key in (
+                RAFT_PASSWORD_KEY,
+                PATRONI_PASSWORD_KEY,
+            ):
+                if self.charm.get_secret(APP_SCOPE, key) is None:
+                    self.charm.set_secret(APP_SCOPE, key, new_password())
+
```

Comment:
Is this the correct place for generating the passes?

Reply:
^ @marceloneppel

Reply:
Yeah, it's ok to put it here.